flow/connectors/mongo/cdc.go

```diff
@@ -26,8 +26,8 @@ type Namespace struct {
 }
 
 type ChangeEvent struct {
-	DocumentKey   map[string]any `bson:"documentKey,omitempty"`
-	FullDocument  map[string]any `bson:"fullDocument,omitempty"`
+	DocumentKey   bson.D         `bson:"documentKey,omitempty"`
+	FullDocument  bson.D         `bson:"fullDocument,omitempty"`
```

Comment:
I changed this to bson.D so cdc's document type is consistent with initial load, otherwise I have to implement redundant logic for map[string]any datatype in the BsonExtension implementation. 

Also realized that bson decoding libary can deserialize to either `bson.D` or `bson.M`. 

Not sure if there's a huge difference in performance between the two (given bson.D is ordered), we can change this to bson.M later if it's faster.